lib/updatePlist.js

```diff
@@ -45,9 +44,39 @@ var getPlistConfigValue = function(plistConfig, manifest, key) {
   return value;
 };
 
+
+var replaceObjectRecursively = function (plistObj, manifest, plistConfig, key) {
+  var value;
+
+  Object.keys(plistConfig).forEach(function (nested_key) {
+    var config_val = plistConfig[nested_key];
+
+    if (typeof config_val === 'object' && !(config_val instanceof Array)) {
+      key.push(nested_key);
+      replaceObjectRecursively(plistObj, manifest, plistConfig[nested_key], key);
+    } else {
+      value = getPlistConfigValue(plistConfig, manifest, nested_key);
+      obj.setVal(plistObj, key.concat(nested_key).join('.'), value);
+      return;
+    }
+    key.pop();
```

Comment:
i think we can move this statement to above if block

Reply:
Then the key value would be empty at first and we don't want it.